### PR TITLE
fix(connect): fix signatureValue of x509certificate

### PR DESCRIPTION
**Pull request by szymonlesisz**
## Description

Fixing malformed signatures produced by optiga
https://github.com/trezor/trezor-firmware/issues/3411

### CODE REVIEW

File: packages/connect/src/api/firmware/x509certificate.ts [hunk 1]
```diff
@@ -98,13 +98,61 @@ const derBitStringValue = (byteArray: Uint8Array) => ({
     bytes: byteArray.subarray(1),
 });
 
+export const parseSignature = (byteArray: Uint8Array) => {
+    const { unusedBits, bytes } = derBitStringValue(byteArray);
+    const asn1 = derToAsn1(bytes);
+
+    if (asn1.cls !== 0 || asn1.tag !== 16 || !asn1.structured) {
+        throw new Error('Bad SPKI. Not a SEQUENCE.');
```
Comment (andrewkozlik) [inline packages/connect/src/api/firmware/x509certificate.ts]: Copy-paste error. SPKI stands for "SubjectPublicKeyInfo", this should say "Bad signature. Not a SEQUENCE."

File: packages/connect/src/api/firmware/__tests__/verifyAuthenticityProof.test.ts [hunk 2]
```diff
@@ -181,3 +182,38 @@ describe('firmware/verifyAuthenticityProof', () => {
         expect(verify.error).toBe('CA_PUBKEY_NOT_FOUND');
     });
 });
+
+describe('firmware/x509certificate', () => {
```
Comment (andrewkozlik) [inline packages/connect/src/api/firmware/__tests__/verifyAuthenticityProof.test.ts:188]: I'd add a comment somewhere here saying that Optiga may produce a malformed signature with probability 1 in 256. These test vectors verify that we are able to correctly reencode the malfomed signature into a valid DER encoding.

Comment (szymonlesisz) [reply packages/connect/src/api/firmware/__tests__/verifyAuthenticityProof.test.ts:188]: done [add6e85](https://github.com/trezor/trezor-suite/pull/10033/commits/add6e8563d725f7528cd093cd279e5be03d7194f)

File: packages/connect/src/api/firmware/x509certificate.ts (hunk 1 above)
Comment (szymonlesisz) [reply packages/connect/src/api/firmware/x509certificate.ts]: done [03d0aac](https://github.com/trezor/trezor-suite/pull/10033/commits/03d0aac57c0a1c4d98a6de68448826379ef5195f)

**Review by andrewkozlik — APPROVED** [review]
utACK

**Pull request merged**